tests/e2e/util/kubectl/kubectl.go

```diff
@@ -129,18 +129,31 @@ func (k Kubectl) ApplyString(yamlString string) error {
 
 // Apply applies the given yaml file to the cluster
 func (k Kubectl) Apply(yamlFile string) error {
-	err := k.ApplyWithLabels(yamlFile, "")
-	return err
+	return k.applyWithOptions("-f " + yamlFile)
 }
 
 // ApplyWithLabels applies the given yaml file to the cluster with the given labels
 func (k Kubectl) ApplyWithLabels(yamlFile, label string) error {
-	cmd := k.build(" apply " + labelFlag(label) + " -f " + yamlFile)
+	return k.applyWithOptions(labelFlag(label) + " -f " + yamlFile)
+}
+
+// ApplyKustomize applies the given kustomization file to the cluster
+func (k Kubectl) ApplyKustomize(kustomizationDirectory string) error {
+	return k.applyWithOptions("-k " + kustomizationDirectory)
+}
```

Comment:
```suggestion
func (k Kubectl) ApplyKustomize(appName string) error {
	return k.applyWithOptions("-k " + common.GetKustomizeDir(appName))
}
```

Since you never call `ApplyKustomize` without getting the directory name, I suggest to use it like this.

Also I'd put `GetKustomizeDir` inside this file and make it a private function, as it's more of an implementation detail.

If you worry that the function name is too generic, perhaps we can call it `ApplyKustomizedSample` or something to make sure callers understand it's specific to samples.

WDYT

Reply:
Yeah I think it will be better in this way and `ApplyKustomizedSample` is a lot more explicit on what we are doing